From: Raul Rodrigo Segura <raurodse@gmail.com>
Date: Mon, 18 Jun 2018 13:28:56 +0200
Subject: remove wrong path

---
 src/libcalamaresui/modulesystem/PythonQtViewModule.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libcalamaresui/modulesystem/PythonQtViewModule.cpp b/src/libcalamaresui/modulesystem/PythonQtViewModule.cpp
index 88b0c20..858097e 100644
--- a/src/libcalamaresui/modulesystem/PythonQtViewModule.cpp
+++ b/src/libcalamaresui/modulesystem/PythonQtViewModule.cpp
@@ -29,7 +29,7 @@
 #include "JobQueue.h"
 
 #include <PythonQt.h>
-#include <extensions/PythonQt_QtAll/PythonQt_QtAll.h>
+#include <PythonQt_QtAll.h>
 
 #include <QDir>
 #include <QPointer>
