Description: Fix default desktop environment variable in displaymanager.
Author: Walter Lapchynski <wxl@ubuntu.com>
--- a/calamares-3.2.1/src/modules/displaymanager/main.py
+++ b/calamares-3.2.1/src/modules/displaymanager/main.py
@@ -398,7 +398,7 @@ def run():
     username = libcalamares.globalstorage.value("autologinUser")
     root_mount_point = libcalamares.globalstorage.value("rootMountPoint")
 
-    if "default_desktop_environment" in libcalamares.job.configuration:
+    if "defaultDesktopEnvironment" in libcalamares.job.configuration:
         entry = libcalamares.job.configuration["defaultDesktopEnvironment"]
         default_desktop_environment = DesktopEnvironment(
             entry["executable"], entry["desktopFile"]
